File: src/HttPlaceholder.Models/StubResponseModel.cs
using System.Collections.Generic;
using YamlDotNet.Serialization;

namespace HttPlaceholder.Models
{
   public class StubResponseModel
   {
      [YamlMember(Alias = "statusCode")]
      public int? StatusCode { get; set; }

      [YamlMember(Alias = "text")]
      public string Text { get; set; }

      [YamlMember(Alias = "base64")]
      public string Base64 { get; set; }

      [YamlMember(Alias = "file")]
      public string File { get; set; }

      [YamlMember(Alias = "headers")]
      public IDictionary<string, string> Headers { get; set; }

      [YamlMember(Alias = "extraDuration")]
      public int? ExtraDuration { get; set; }

      [YamlMember(Alias = "json")]
      public string Json { get; set; }

      [YamlMember(Alias = "xml")]
      public string Xml { get; set; }

      [YamlMember(Alias = "html")]
      public string Html { get; set; }

      [YamlMember(Alias = "temporaryRedirect")]
      public string TemporaryRedirect { get; set; }

      [YamlMember(Alias = "permanentRedirect")]
      public string PermanentRedirect { get; set; }
   }
}

File: gui/src/domain/stub/stub-response-model.ts
import type { HashMap } from "@/domain/hash-map";
import type { StubResponseReverseProxyModel } from "@/domain/stub/stub-response-reverse-proxy-model";
import type { LineEndingType } from "@/domain/stub/enums/line-ending-type";
import type { StubResponseImageModel } from "@/domain/stub/stub-response-image-model";
import type { StubResponseScenarioModel } from "@/domain/stub/stub-response-scenario-model";

export interface StubResponseModel {
  enableDynamicMode?: boolean;
  statusCode?: number;
  contentType?: string;
  text?: string;
  base64?: string;
  file?: string;
  headers?: HashMap;
  extraDuration?: number;
  json?: string;
  xml?: string;
  html?: string;
  temporaryRedirect?: string;
  permanentRedirect?: string;
  reverseProxy?: StubResponseReverseProxyModel;
  lineEndings?: LineEndingType;
  image?: StubResponseImageModel;
  scenario?: StubResponseScenarioModel;
  abortConnection?: boolean;
}
